std/functional.d

```diff
@@ -747,6 +741,35 @@ template partial(alias fun, alias arg)
             }
         }
     }
+    else static if (!isCallable!fun)
+    {
+        static assert(false, "Cannot apply partial to a non-callable '" ~ fun.stringof ~ "'.");
+    }
+    else // Assume fun is callable and uniquely defined.
+    {
+        static if (Parameters!fun.length == 0)
```

Comment:
I think this case is unnecessary. If they don't pass any arguments, just alias `partial` to `fun`.

Reply:
This check states that the function to be partially invoked should have arguments. I will update the error message to reflect this.
As far as I see there is no support for partially invoking functions with multiple partial arguments.